splunk_add_on_ucc_framework/entity/index_entity.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2024 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Optional, Dict, Any
+
+from splunk_add_on_ucc_framework.entity.entity import Entity
+
+
+class IndexEntity(Entity):
+    def long_form(self) -> Dict[str, Any]:
+        definition = {
+            "type": "singleSelect",
+            "field": self["field"],
+            "label": self["label"],
+            "defaultValue": "default",
+            "options": {
+                "endpointUrl": "data/indexes?search=isInternal=0+disabled=0",
+                "denyList": "^_.*$",
+                "createSearchChoice": True,
+            },
+            "validators": [
+                {
+                    "type": "regex",
+                    "errorMsg": "Index names must begin with a letter or a number "
+                    "and must contain only letters, numbers, underscores or hyphens.",
+                    "pattern": r"^[a-zA-Z0-9][a-zA-Z0-9\\_\\-]*$",
```

Comment:
Did you craft this regex yourself? Would it be possible to find a link to Splunk docs to confirm that it is indeed a regex for index?